browser/resources/settings/brave_sync_page/brave_sync_page.js

```diff
@@ -85,30 +84,24 @@ Polymer({
    * @param {?SyncStatus} syncStatus
    * @private
    */
-  handleSyncStatus_: function(syncStatus) {
-    console.debug('Sync Status Update', syncStatus)
+  handleSyncStatus_: async function(syncStatus) {
     this.syncStatus_ = syncStatus;
-  },
 
-  /**
-   * Handler for when the sync preferences are updated.
-   * @private
-   */
-  handleSyncPrefsChanged_: async function(syncPrefs) {
-    // Enforce encryption
-    if (this.syncStatus_ && !this.syncStatus_.firstSetupInProgress) {
-      if (!syncPrefs.encryptAllData) {
-        const syncCode = await this.braveBrowserProxy_.getSyncCode()
-        syncPrefs.encryptAllData = true;
-        syncPrefs.setNewPassphrase = true;
-        syncPrefs.passphrase = syncCode;
-        await this.browserProxy_.setSyncEncryption(syncPrefs)
-      } else if (syncPrefs.passphraseRequired) {
-        const syncCode = await this.braveBrowserProxy_.getSyncCode()
-        syncPrefs.setNewPassphrase = false;
-        syncPrefs.passphrase = syncCode;
-        await this.browserProxy_.setSyncEncryption(syncPrefs)
-      }
+    // Cleanup is_encryption_set_ for re-enabling sync
+    if (!this.syncStatus_.isEngineInitialized &&
+      this.syncStatus_.firstSetupInProgress && this.is_encryption_set_) {
+      this.is_encryption_set_ = false
+    }
```

Comment:
I don't think we need this if we provide default value correctly as I suggested above